### PR TITLE
Better html mails

### PR DESCRIPTION
**Pull request by Gomez**
Update HTMLPurifier, add CSSTidy. Add styles tag back on html mails

### CODE REVIEW

File: lib/Service/Html.php [hunk 1]
```diff
@@ -129,11 +133,22 @@ public function sanitizeHtmlMailBody($mailBody, array $messageParameters, Closur
...
 
 		$result = $purifier->purify($mailBody);
 		// eat xml parse errors within HTMLPurifier
 		libxml_clear_errors();
+
+		// Add back the style tag
+		$styles = $purifier->context->get('StyleBlocks');
+		if ($styles) {
+			$result = implode("\n", array(
+				'<style type="text/css">',
```
Comment (ChristophWurst) [inline lib/Service/Html.php]: @LukasReschke anything scary about this? Is it safe? :)

Comment (Gomez) [reply lib/Service/Html.php]: AFAIK it runs through CSSTidy, should be good :)

Comment (Gomez) [reply lib/Service/Html.php]: There is a warning: http://htmlpurifier.org/live/configdoc/plain.html at "Filter.ExtractStyleBlocks". @LukasReschke imagecrash attack?

Comment (ChristophWurst) [reply lib/Service/Html.php]: according to @LukasReschke this change is okay as long as we're iframing the message.

**Pull request merged**